sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -477,16 +581,22 @@ def __init__(
     self._key_coder = key_coder
     self._window_coder = window_coder
     self._timer_specs = timer_specs
-    self._timer_receivers = None
-    self._all_states = {}
+    self._timer_receivers = None  # type: Optional[Dict[str, operations.ConsumerSet]]
+    self._all_states = {}  # type: Dict[tuple, Union[SynchronousBagRuntimeState, SynchronousSetRuntimeState, CombiningValueRuntimeState]]
```

Comment:
I could replace this with `Dict[tuple, userstate.AccumulatingRuntimeState]` but in order for this to work I need to add `_commit` to the abstract methods on this `AccumulatingRuntimeState`, because this method is expected to exist:

```python
class FnApiUserStateContext(userstate.UserStateContext):
  def __init__(self):
    ...
    self._all_states = {}  # type: Dict[tuple, userstate.AccumulatingRuntimeState]

  ...

  def commit(self):
    # type: () -> None
    for state in self._all_states.values():
      state._commit()
```

Reply:
This is still unresolved.  Shall I add `commit` abstract method to `AccumulatingRuntimeState`, and rename `_commit` to `commit` on the concrete implementations?